IdentityService/API/Controller.cs

```diff
@@ -3,16 +3,39 @@
 public class Controller : ControllerBase
 {
     private readonly IUserService _userService;
+    private readonly JwtConfiguration _jwtConfiguration;
 
-    public Controller(IUserService userService) => _userService = userService;
+    public Controller(IUserService userService, JwtConfiguration jwtConfiguration)
+    {
+        _userService = userService;
+        _jwtConfiguration = jwtConfiguration;
+    }
 
     [HttpPost("login")]
-    public async Task<string> Login([FromBody] LoginDto login)
+    public async Task<JwtDto> Login([FromBody] LoginDto login)
     {
-        var permissions = await _userService.Login(login);
-        return new JwtSecurityTokenHandler().WriteToken(new JwtSecurityToken(
-            claims: permissions.Select(permission => new Claim("Permissions", ((int)permission).ToString())),
-            signingCredentials: new(new SymmetricSecurityKey("TESTTESTTESTTEST"u8.ToArray()), SecurityAlgorithms.HmacSha256)
-        ));
+        var user = await _userService.Login(login);
+        var identityTokenClaims = user.Permissions.Select(permission => new Claim("Permissions", ((int)permission).ToString())).ToArray();
+        var refreshTokenClaims = new[] { new Claim("Id", user.Id.ToString()) };
+        return new()
+        {
+            IdentityToken = Jwt.Generate(identityTokenClaims, TimeSpan.FromSeconds(_jwtConfiguration.IdentityTokenExpiresInSeconds)),
+            RefreshToken = Jwt.Generate(refreshTokenClaims, TimeSpan.FromSeconds(_jwtConfiguration.RefreshTokenExpiresInSeconds)),
+        };
     }
+
+    [Authorize]
+    [HttpGet("verify")]
+    public void Verify() { }
+
+    [Authorize]
+    [HttpGet("refresh")]
+    public async Task<string> Refresh()
```

Comment:
Shouldn't user provide refresh token to be able to refresh token?

Reply:
See reply above, we differentiate between the two tokens by the Id claim